### PR TITLE
Travis: Add an rpi-dispmanx build on 18.04 armhf

### PR DESCRIPTION
**Pull request by RAOF**

### CODE REVIEW

**Comment by RAOF** [comment]
bors try

File: .travis.yml [hunk 1]
```diff
@@ -68,7 +69,7 @@ before_install:
 - popd
 
 script:
-- ${SPREAD_PATH}/spread -v $SYSTEM:...:$VARIANT
+- ${SPREAD_PATH}/spread -v $SYSTEM:spread/build/$TASK:$VARIANT
```
Comment (Saviq) [inline .travis.yml]: This split looks artificial now, we could just have a single variable, explicitly listing the tasks in the `matrix` above?

Note there's more `SYSTEM` variables below, in the `sanitize` stage.

**Comment by RAOF** [comment]
We now have the first cut of a rpi-userland snap, which is enough to build against and can be expanded until it's also enough to *run* against (*ideally* by becoming a “gl interface” provider, but…)

**Comment by Saviq** [comment]
But also:
```
[ 11%] Building CXX object src/platforms/rpi-dispmanx/CMakeFiles/mirplatformgraphicsrpidispmanxobjects.dir/buffer_allocator.cpp.o
cd /tmp/tmp.UmBIs4gInC/src/platforms/rpi-dispmanx && /usr/bin/arm-linux-gnueabihf-g++  -DEGL_NO_X11 -DLOG_NDEBUG=1 -DLTTNG_UST_HAVE_SDT_INTEGRATION -DMESA_EGL_NO_X11_HEADERS -DMIR_SERVER_EGL_OPENGL_API=EGL_OPENGL_ES_API -DMIR_SERVER_EGL_OPENGL_BIT=EGL_OPENGL_ES2_BIT -DMIR_SERVER_GLEXT_H="<GLES2/gl2ext.h>" -DMIR_SERVER_GL_H="<GLES2/gl2.h>" -DMIR_VERSION_MAJOR=1 -DMIR_VERSION_MICRO=0 -DMIR_VERSION_MINOR=6 -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE -I/spread/mir/include/core -I/spread/mir/include/common -I/spread/mir/include/cookie -I/snap/rpi-userland/current/opt/vc/include -I/snap/rpi-userland/current/opt/vc/include/interface/vmcs_host/linux -I/snap/rpi-userland/current/opt/vc/include/interface/vcos/pthreads -I/spread/mir/include/platform -I/spread/mir/include/renderers/gl -I/spread/mir/include/renderers/sw -I/spread/mir/src/platforms/common/server -I/spread/mir/src/include/platform -I/spread/mir/src/include/common  -pthread -g -std=c++14 -Wall -fno-strict-aliasing -pedantic -Wnon-virtual-dtor -Wextra -fPIC -Werror -Wno-psabi -g   -Wno-error=variadic-macros -std=c++14 -o CMakeFiles/mirplatformgraphicsrpidispmanxobjects.dir/buffer_allocator.cpp.o -c /spread/mir/src/platforms/rpi-dispmanx/buffer_allocator.cpp
In file included from /snap/rpi-userland/current/opt/vc/include/interface/vcos/vcos.h:144:0,
                 from /snap/rpi-userland/current/opt/vc/include/interface/vmcs_host/vc_dispmanx.h:33,
                 from /spread/mir/src/platforms/rpi-dispmanx/buffer_allocator.cpp:24:
/snap/rpi-userland/current/opt/vc/include/interface/vcos/vcos_logging.h:234:39: warning: ISO C++ does not permit named variadic macros [-Wvariadic-macros]
 #  define _VCOS_LOG_X(cat, _level, fmt...)   do { if (vcos_is_log_enabled(cat,_level)) vcos_log_impl(cat,_level,fmt); } while (0)
                                       ^~~
In file included from /spread/mir/src/platforms/rpi-dispmanx/rendering_platform.cpp:22:0:
/spread/mir/include/common/mir/module_deleter.h: In instantiation of âmir::UniqueModulePtr<Type> mir::{anonymous}::make_module_ptr(Args&& ...) [with Type = mir::graphics::rpi::BufferAllocator; Args = {const mir::graphics::Display&}; mir::UniqueModulePtr<Type> = std::unique_ptr<mir::graphics::rpi::BufferAllocator, mir::ModuleDeleter<mir::graphics::rpi::BufferAllocator> >]â:
/spread/mir/src/platforms/rpi-dispmanx/rendering_platform.cpp:35:61:   required from here
/spread/mir/include/common/mir/module_deleter.h:100:34: error: invalid new-expression of abstract class type âmir::graphics::rpi::BufferAllocatorâ
     return UniqueModulePtr<Type>(new Type(std::forward<Args>(args)...), std::move(deleter));
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /spread/mir/src/platforms/rpi-dispmanx/rendering_platform.cpp:28:0:
/spread/mir/src/platforms/rpi-dispmanx/buffer_allocator.h:60:7: note:   because the following virtual functions are pure within âmir::graphics::rpi::BufferAllocatorâ:
 class BufferAllocator :
       ^~~~~~~~~~~~~~~
In file included from /spread/mir/src/platforms/rpi-dispmanx/buffer_allocator.h:24:0,
                 from /spread/mir/src/platforms/rpi-dispmanx/rendering_platform.cpp:28:
/spread/mir/include/platform/mir/graphics/wayland_allocator.h:50:18: note: 	virtual std::shared_ptr<mir::graphics::Buffer> mir::graphics::WaylandAllocator::buffer_from_shm(wl_resource*, std::shared_ptr<mir::Executor>, std::function<void()>&&)
     virtual auto buffer_from_shm(
                  ^~~~~~~~~~~~~~~
src/platforms/rpi-dispmanx/CMakeFiles/mirplatformgraphicsrpidispmanxobjects.dir/build.make:134: recipe for target 'src/platforms/rpi-dispmanx/CMakeFiles/mirplatformgraphicsrpidispmanxobjects.dir/rendering_platform.cpp.o' failed
make[2]: *** [src/platforms/rpi-dispmanx/CMakeFiles/mirplatformgraphicsrpidispmanxobjects.dir/rendering_platform.cpp.o] Error 1
```

**Comment by RAOF** [comment]
Hah! Because we merged the shm-from-platform branch without me noticing I failed to do the rpi bit! (Which was exactly what this PR was meant to be about 😀)

**Comment by RAOF** [comment]
Merging #1190 will verify this works :)

**Comment by AlanGriffiths** [comment]
> Yeah
> 
> bors r+

Hmm, this is the second "bors r+" that bors seems to have missed today. (C.f. #1205)

**Comment by AlanGriffiths** [comment]
bors r=Saviq

**Pull request merged**